imagedephi/utils/tiff.py

```diff
@@ -13,3 +41,59 @@ def get_tiff_tag(tag_name: str) -> tifftools.TiffTag:
         if tag_name in tag_set:
             return tag_set[tag_name]
     return tifftools.constants.get_or_create_tag(tag_name)
+
+
+def get_associated_image_svs(image_path: Path, image_key: str) -> IFD | None:
+    """Given a path to an SVS image, return the IFD for a given associated label or macro image."""
+    if image_key not in ["macro", "label"]:
+        raise ValueError("image_key must be one of macro, label")
+
+    image_info = tifftools.read_tiff(image_path)
+    ifds = image_info["ifds"]
+
+    image_description_tag = tifftools.constants.Tag["ImageDescription"]
+    newsubfiletype_tag = tifftools.constants.Tag["NewSubfileType"]
+
+    if "aperio" not in str(ifds[0]["tags"][image_description_tag.value]["data"]).lower():
+        # raise ValueError(f"{image_path} is not an svs image")
+        return None
+
+    for ifd in iter_ifds(ifds):
+        if image_description_tag.value in ifd["tags"]:
+            if image_key in str(ifd["tags"][image_description_tag.value]["data"]):
+                return ifd
+        if newsubfiletype_tag.value in ifd["tags"]:
+            newsubfiletype = ifd["tags"][newsubfiletype_tag.value]["data"][0]
+            if int(newsubfiletype) & 8 and image_key == "macro":
+                return ifd
+    return None
+
+
+def get_ifd_for_thumbnail(image_path: Path) -> IFD | None:
+    """Given a path to a TIFF image, return the IFD for the lowest resolution tiled image."""
+    image_info = tifftools.read_tiff(image_path)
+
+    min_width = float("inf")
+    lowest_res_ifd = None
+    for ifd in iter_ifds(image_info["ifds"]):
+        # We are interested in the lowest res tiled image.
+        if tifftools.Tag.TileWidth.value not in ifd["tags"]:
+            continue
+        # PIL can only read JPEG-compressed tiff images
+        if ifd["tags"][tifftools.Tag.Compression.value] != tifftools.constants.Compression.JPEG:
+            continue
```

Comment:
@manthey Are there other compression formats that we might come across that PIL would support?